client/js/contextMenu.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+const $ = require("jquery");
+const templates = require("../views");
+let contextMenu, contextMenuContainer;
+
+module.exports = class ContextMenu {
+	constructor(contextMenuItems, contextMenuActions, selectedElement, event) {
+		this.contextMenuItems = contextMenuItems;
+		this.contextMenuActions = contextMenuActions;
+		this.selectedElement = selectedElement;
+		this.event = event;
+
+		contextMenuContainer = $("#context-menu-container");
+		contextMenu = $("#context-menu");
+	}
+
+	show() {
+		showContextMenu(this.contextMenuItems, this.selectedElement, this.event);
+		this.bindEvents();
+		return false;
+	}
+
+	bindEvents() {
+		const contextMenuActions = this.contextMenuActions;
+
+		contextMenuActions.execute = (id, ...args) => contextMenuActions[id] && contextMenuActions[id](...args);
+
+		contextMenu.find(".context-menu-item").on("click", function() {
+			const $this = $(this);
+			const itemData = $this.data("data");
+			const contextAction = $this.data("action");
+			contextMenuActions.execute(contextAction, itemData);
+		});
+	}
+};
+
+function showContextMenu(contextMenuItems, selectedElement, event) {
```

Comment:
I believe you wanted to make some code changes, you might as well bring the content of this function back into the method with the same name in the class above, no?

Reply:
I guess so...I kinda prefer splitting out the actual code to show the context menu from the logic around binding the events to it, but if you want this in, then sure.

Reply:
I mean, I'm not _demanding_ it, if you don't like that, that's fine, I just don't see the point of the extra level of indirection in this case, but maybe I'm just missing something here or in what you're planning to add/change soon after. Your call 🤷‍♂️

Reply:
I mean I won't change it unless you ask me to because I prefer this way :-P but I'm not so attached to it that I will argue too much about it.

The reason being is that I feel like the `open()` function has 2 logical blocks, this is 1 of the logical blocks, and having this logical block inlined would make that function a little harder to read. I like to split things up into smaller blocks because it makes it easier to structure and read.

Reply:
Sure, doesn't matter much anyway.